### PR TITLE
Added test for Socket Address

### PR DESCRIPTION
**Pull request by zrthxn**
This still doesnt include `folly::MacAddress` because the formatter doesn't work for that.

### CODE REVIEW

File: src/address/address.cpp [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #include <iostream>
 #include <string>
 #include <folly/IPAddress.h>
+#include <folly/MacAddress.h>
```
Comment (a20012251) [inline src/address/address.cpp:4]: delete this if you are not using it

File: src/address/address.cpp [hunk 2]
```diff
@@ -14,12 +16,24 @@ void printIP6(std::string ip6_str) {
   std::cout << ip6 << std::endl;
 }
 
+void printSocket(std::string host_str, uint16_t port) {
```
Comment (a20012251) [inline src/address/address.cpp:19]: Actually, this test should be in a completely different folder and test file: src/socket
The idea is that each test is minimal and only test one single thing. That gives us better signals if something goes wrong

**Review by a20012251 — CHANGES_REQUESTED** [review]
Read the comments

File: src/address/address.cpp (hunk 1 above)
Comment (zrthxn) [reply src/address/address.cpp]: done

**Comment by zrthxn** [comment]
Done the changes.

File: src/address/address.cpp [hunk 3]
```diff
@@ -1,25 +1,21 @@
...
 
-void printIP4(std::string ip4_str) {
-  IPAddress ip4(ip4_str);
-  std::cout << ip4 << std::endl;
-}
-
-void printIP6(std::string ip6_str) {
-  IPAddress ip6(ip6_str);
-  std::cout << ip6 << std::endl;
+void printIP(std::string ip_str) {
```
Comment (a20012251) [inline src/address/address.cpp:6]: good simplification

**Review by a20012251 — APPROVED** [review]

**Pull request merged**